tests/kernel/mutex/mutex_api/src/test_mutex_apis.c

```diff
@@ -387,6 +387,52 @@ ZTEST_USER(mutex_api_1cpu, test_mutex_priority_inheritance)
 	k_msleep(TIMEOUT+1000);
 }
 
+static void tThread_mutex_lock_should_fail(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	struct k_mutex *mutex = (struct k_mutex *)p1;
+
+	zassert_equal(-EAGAIN, k_mutex_lock(mutex, K_MSEC(TIMEOUT)), NULL);
+}
+
+/**
+ * @brief Test fix for subtle race during priority inversion
+ *
+ * - A low priority thread (Tlow) locks mutex A.
+ * - A high priority thread (Thigh) blocks on mutex A, boosting the priority
+ *   of Tlow.
+ * - Thigh times out waiting for mutex A.
+ * - Before Thigh has a chance to execute, Tlow unlocks mutex A (which now
+ *   has no owner) and drops its own priority.
+ * - Thigh now gets a chance to execute and finds that it timed out, and
+ *   then enters the block of code to lower the priority of the thread that
+ *   owns mutex A (now nobody).
+ * - Thigh tries to the dereference the owner of mutex A (which is nobody,
+ *   and thus it is NULL). This leads to an exception.
+ *
+ * @ingroup kernel_mutex_tests
+ *
+ * @see k_mutex_lock()
+ */
+ZTEST(mutex_api_1cpu, test_mutex_timeout_race_during_priority_inversion)
+{
+	k_mutex_init(&mutex);
+
+	/* align to tick boundary */
+	k_sleep(K_TICKS(1));
+	k_thread_create(&tdata, tstack, K_THREAD_STACK_SIZEOF(tstack),
+			tThread_mutex_lock_should_fail, &mutex, NULL, NULL, 8,
+			K_USER | K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_mutex_lock(&mutex, K_FOREVER);
+
+	k_sleep(K_MSEC(TIMEOUT));
```

Comment:
If priority inheritance is kicking in, then both the helper and the current thread should be at the same priority. They are both likely to wake on the same tick. The test as written is assuming that the current thread is going to be executed first. Such an assumption might not hold--it is highly dependent upon the implementation of the scheduler and which thread is got to the "head" of the ready queue first.

To work around this, I have two ideas...
1. If available, use an absolute timeout for both the helper and current thread to guarantee that both threads will expire on the same tick. 
2. Before the current thread sleeps, explicitly bump its priority even higher. This should help guarantee that it will always execute before the helper thread.

Reply:
@peter-mitsis - I could create an arbitrary absolute tick value in the current thread and pass that as a parameter to the helper thread.

One option is making that a Kconfig value local to the testsuite.

The main concern I have with that is that it could be another one of those racey values that we just have to keep increasing over time. Is there a happy medium between long-enough to allow the thread to start and short enough to not be considered "too long for a test"?